merkledag/merkledag.go

```diff
@@ -460,16 +443,24 @@ func EnumerateChildrenAsync(ctx context.Context, ds DAGService, c *cid.Cid, visi
 			if inProgress == 0 && next == nil {
 				return nil
 			}
-		case nc := <-out:
-			if nc.Err != nil {
-				return nc.Err
+		case l := <-out:
+			n, err := ds.Get(ctx, l)
+			if err != nil {
+				return err
 			}
 
 			if next == nil {
-				next = nc.Node
-				send = feed
+				if len(n.Links()) > 0 {
```

Comment:
This can be cleaned up a little bit, Lets add all the new links to the array first, then check for next being nil and handle that case separately.